message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -1695,18 +1695,17 @@
       <field type="float" name="increment">OSD parameter increment.</field>
     </message>
     <message id="11037" name="OBSTACLE_DISTANCE_3D">
-      <wip/>
-      <!-- This message is work-in-progress and it can therefore change, and should NOT be used in stable production environments -->
       <description>Obstacle located as a 3D vector.</description>
       <field type="uint32_t" name="time_boot_ms" units="ms">Timestamp (time since system boot).</field>
       <field type="uint8_t" name="sensor_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
       <field type="uint8_t" name="frame" enum="MAV_FRAME">Coordinate frame of reference.</field>
-      <field type="uint16_t" name="obstacle_id" instance="true"> Unique ID given to each obstacle so that its movement can be tracked. Use UINT16_MAX if object ID is unknown or cannot be determined.</field>
-      <field type="float" name="x" units="m"> X position of the obstacle.</field>
-      <field type="float" name="y" units="m"> Y position of the obstacle.</field>
-      <field type="float" name="z" units="m"> Z position of the obstacle.</field>
+      <field type="uint16_t" name="obstacle_id" instance="true">Unique ID given to each obstacle so that its movement can be tracked. Use UINT16_MAX if object ID is unknown or cannot be determined.</field>
+      <field type="float" name="x" units="m">X position of the obstacle.</field>
+      <field type="float" name="y" units="m">Y position of the obstacle.</field>
+      <field type="float" name="z" units="m">Z position of the obstacle.</field>
       <field type="float" name="min_distance" units="m">Minimum distance the sensor can measure.</field>
       <field type="float" name="max_distance" units="m">Maximum distance the sensor can measure.</field>
+      <field type="uint8_t" name="orientation" enum="MAV_SENSOR_ORIENTATION">Direction the sensor faces. downward-facing: ROTATION_PITCH_270, upward-facing: ROTATION_PITCH_90, backward-facing: ROTATION_PITCH_180, forward-facing: ROTATION_NONE, left-facing: ROTATION_YAW_90, right-facing: ROTATION_YAW_270</field>
```

Comment:
Perhaps this should go up next to `MAV_FRAME`?

We also need to clarify how `orientation` interacts with `MAV_FRAME`.  Which is probably, "we ignore all but FRD so orientation is always vs FRD".

Reply:
Should we specify here which frames we support?  isn't that a bit too AP specific?  or even AP Version specific?